src/viking.js

```diff
@@ -1,11 +1,92 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor(health, strength) {
+        this.health = health
+        this.strength = strength
+    }
 
-// Viking
-class Viking {}
+    attack() {
+     return this.strength
+    }
+    receiveDamage(damage) {
+     this.health -= damage   
+    }
+}
 
-// Saxon
-class Saxon {}
+class Viking extends Soldier {
+    constructor(name, health, strength) {
+        super(health, strength);
+        this.name = name
+    }
+    receiveDamage(damage) {
+        this.health -= damage
+        let dead = this.health < 0
+        if (Viking !== dead) {
+            console.log(`${Viking.name} has received ${damage} points of damage`)
+        } else {
+            console.log(`${Viking.name} has died in act of combat`)
+        }
+    }
+    battleCry() {
+        return "Odin Owns You All!"
+    }
+}
+
+class Saxon extends Soldier {
+    receiveDamage(damage) {
+        Saxon.health -= damage
```

Comment:
Aqui na linha 37, usaremos o this, que é o objeto Saxon, igualzinho vc fez na class Viking. 
this.health -= damage, ficando assim tá bom?
No if seguinte, também, colocamos this e não a class Saxon, nesse caso; o attack( ), ele não precisa aparecer aqui na class Saxon, pq vc já está dizendo que é uma herança de Soldier, e que esse método attack ( )é igual nas duas classes. Usando o this novamente, ficaria igual, logo...não usamos Saxon e reaproveitamos o código então;